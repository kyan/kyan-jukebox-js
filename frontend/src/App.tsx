import React from 'react'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import { Toaster } from 'react-hot-toast'
import ErrorBoundary from 'components/error-boundary'
import jukeboxMiddleware from 'middleware/jukebox-middleware'
import jukeboxApp from 'reducers'
import DashboardContainer from 'containers/dashboard-container'

const store = createStore(jukeboxApp, applyMiddleware(jukeboxMiddleware))

const App = () => {
  return (
    <Provider store={store}>
      <Toaster
        position='bottom-left'
        toastOptions={{
          duration: 4000,
          style: {
            background: '#363636',
            color: '#fff'
          }
        }}
      />
      <ErrorBoundary>
        <DashboardContainer />
      </ErrorBoundary>
    </Provider>
  )
}

export default App
