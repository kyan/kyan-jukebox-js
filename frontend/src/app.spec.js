import React from 'react'
import { render } from '@testing-library/react'
import App from './App'

describe('App', () => {
  it('renders without crashing', () => {
    const { container } = render(<App />)
    expect(container).toBeInTheDocument()
  })
})
