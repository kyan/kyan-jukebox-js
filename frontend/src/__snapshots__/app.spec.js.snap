// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Container
    breakpoint={768}
    isMobile={true}
  />
  <ContextProvider>
    <ErrorBoundary>
      <DragDropContext(DashboardContainer) />
    </ErrorBoundary>
  </ContextProvider>
</Provider>
`;
